### ISSUE TITLE
How to share files between bazel test sandbox and docker container(running in root) with proper permissions and cleanup?

### ISSUE THREAD

**Issue by nianhsuantsai-db** [bazelbuild/bazel#18164]
### Description of the bug:

I am launching a docker container within bazel tests to test services running in the docker container. And in my scenario, I need a shared storage location to which both my test process and the docker service under test can access. Our initial attempt was to create a shared directory in the bazel sandbox under [$TEST_TMPDIR](https://bazel.build/reference/test-encyclopedia#test-interaction-filesystem) within the test process and bind mount the shared directory into the docker services. However, since the docker services run with root permission by default, we have two problems:
1. The docker service writes files with root permission to the shared directory. The test process does not have permission to modify these files.
2. The bazel sandbox cleanup fails to clean up the root files left behind. The cleanup problem is similar to this [github issue](https://github.com/bazelbuild/bazel/issues/5211).

**Are there any general solutions to share files between the bazel test and the docker containers while avoiding the permission and cleanup issues?**

Some assumptions here:

1. I have no control over the docker files.
2. The docker services run with root, and the test host process runs with an unprivileged user.

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

TL;DR

- We have a bazel test that launches a docker container.
- The service running in the docker container needs to write files at a location that the test process can later access.
- We mount a location, which is in the sandbox, when launching the container to share the directory in the two environments.
- The service writes the file but runs with root. The test process cannot access these files, and bazel is unable to clean the sandbox with root files left behind.

### Which operating system are you running Bazel on?

Ubuntu 18.04.5 LTS

### Have you found anything relevant by searching the web?

No.

**Comment by Pavank1992** [comment]
Hi @nianhsuantsai-db, It would be helpful to reproduce the issue if you share minimal steps with sample code.
